tests/containers/testenv_validation.pm

```diff
@@ -0,0 +1,39 @@
+# SUSE's openQA tests
+#
+# Copyright 2009-2013 Bernhard M. Wiedemann
+# Copyright 2013-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman engine
+# Summary: Verify Podman nginx container is still running after upgrade
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use containers::common;
+use containers::utils;
+use containers::container_images;
+
+
+sub run {
+    my ($self) = @_;
+    my $unit_name = 'my_nginx';
+    my $container_name = 'nginx-Quadlet';
+    select_serial_terminal;
+
+    my $podman = $self->containers_factory('podman');
+    $self->{podman} = $podman;
+
+
+    # validate the container
+    my $target_version = get_var("TARGET_VERSION", get_required_var("VERSION"));
+    assert_script_run("grep VERSION= /etc/os-release | grep $target_version");
+    validate_script_output("podman ps", qr/$container_name/);
+    validate_script_output("podman container inspect --format='{{.State.Running}}' $container_name", qr/true/);
+    systemctl("is-active $unit_name.service");
+    validate_script_output("curl -s http://localhost:80 | grep title", sub { m/Welcome to the nginx container!/ });
+}
```

Comment:
Missing cleanup call: `$podman->cleanup_system_host();`

Reply:
There cannot be a cleanup here because the state of the containers needs to be preserved. The overall test design is as follows

* Prepare test environment (create containers)
* Validate containers
* Do the migration
* Validate the containers again

Therefore the prepared and validated containers need to persist.